rust/src/smb/smb2_records.rs

```diff
@@ -401,12 +402,31 @@ pub struct Smb2ReadResponseRecord<'a> {
     pub data: &'a[u8],
 }
 
+// parse read/write data. If all is available, 'take' it.
+// otherwise just return what we have. So this may return
+// partial data.
+fn parse_smb2_data<'a>(i: &'a[u8], len: u32)
+    -> IResult<&'a[u8], &'a[u8]>
+{
+    if len as usize > i.len() {
+        IResult::Done(&[], i)
+    } else {
+        do_parse!(i,
+                data: take!(len)
+            >> (data)
+        )
```

Comment:
Minor point: `do_parse!` is useless here, you can just call `take!` (this is faster)
```rust
} else {
    take!(i, len)
}
```